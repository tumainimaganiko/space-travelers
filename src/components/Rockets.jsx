const Rockets = () => (
  <div>Rockets</div>
);

export default Rockets;
