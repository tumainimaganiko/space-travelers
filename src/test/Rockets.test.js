import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import axios from 'axios';
import myRockets from './rocketData';
import Rockets from '../components/Rockets';

jest.mock('axios');

const reducer = (
  state = {
    rockets: { rocket: myRockets },
  },
) => state;

const store = configureStore({ reducer });

describe('Testing the Rocket Component', () => {
  it('Test the rockets data', () => {
    axios.get.mockResolvedValue({ data: myRockets });
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <Rockets />
          </Provider>
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
