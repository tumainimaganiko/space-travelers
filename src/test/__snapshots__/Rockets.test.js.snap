// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Rocket Component Test the rockets data 1`] = `
<div>
  loading
</div>
`;
